src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapter.java

```diff
@@ -34,10 +44,19 @@ public void operateTo(int x, int y) {
 		line.setEndCoordinates(x, y);
 
 		drawController.drawLine(line);
+
+		addCall(new OperateToCommand(x, y));
 	}
 
 	@Override
 	public String toString() {
 		return "2d device simulator - " + name;
 	}
+
+	@Override
+	public void addCall(DriverCommand driverCommand) {
+		for (DriverCallRecorderObserver observer : getDriverCallRecorderObservers()) {
+			observer.recordDriverCall(driverCommand);
```

Comment:
It isn't an observer (recordDriverCall is to specific to be considered an update()) and I am not sure what for is getDriverCallRecorderObservers() collection.